File: ruby/rna-transcription/rna_transcription.rb
module Complement
  COMPLEMENTS = { 'G'=> 'C', 'C'=> 'G', 'T'=> 'A', 'A'=> 'U', '' => ''}

  def self.of_dna(dna)
    dna.split('').map{|n| COMPLEMENTS[n]}.join
  end
end

File: javascript/rna-transcription/rna-transcription.js
export const toRna = (dna) => {
  const complements = {
    'G': 'C',
    'C': 'G',
    'T': 'A',
    'A': 'U'
  } 

  return [...dna].map(n => complements[n]).join('')
};
